### PR TITLE
Init translate

### PR DESCRIPTION
**Pull request by peacebonez**
Checks for a translate condition in the back-end, translates the message, and then stores it in a Conversation instance.

Also fixed a little bug on the front end that now shows avatar's on incoming friend requests.

### CODE REVIEW

File: functions/translate.js [hunk 1]
```diff
@@ -6,6 +6,9 @@ const { Translate } = require('@google-cloud/translate').v2;
 const translate = new Translate({ projectId });
 
 const translateText = async (text, language) => {
+  if (language === 'English') language = 'en';
```
Comment (ThinkTink) [inline functions/translate.js:9]: Eventually we'll probably want to support more languages, and would probably want to pull this out into a constant outside the function. This is fine for now though

File: models/Conversation.js [hunk 2]
```diff
@@ -34,6 +34,7 @@ const ConversationSchema = new Schema({
       text: {
         type: String,
       },
```
Comment (ThinkTink) [inline models/Conversation.js:36]: this needs a type right?

**Review by ThinkTink — COMMENTED** [review]
Looks good but this is only one side? We need to be able to get the translations as well

File: models/Conversation.js (hunk 2 above)
Comment (peacebonez) [reply models/Conversation.js:36]:
```
     translations: {
        type: Object,
        English: {type: String},
        Spanish: {type: String},
        French: {type: String},
      },
```

I'm uncertain how to define a type for the dynamic keys so for now this is what I have.  
Ideally it look something like translations[language]: {type: String}

**Comment by peacebonez** [comment]
> Looks good but this is only one side? We need to be able to get the translations as well

By "get" do you mean loading the data onto the front end.  Because yes I need to alter some of our object structures to make that happen.

**Comment by ThinkTink** [comment]
> > Looks good but this is only one side? We need to be able to get the translations as well
> 
> By "get" do you mean loading the data onto the front end. Because yes I need to alter some of our object structures to make that happen.

Yup! Is the plan to do that in this PR or a separate one?

**Pull request merged**